pallets/baby-liminal/src/lib.rs

```diff
@@ -114,40 +155,56 @@ pub mod pallet {
         #[pallet::call_index(0)]
         #[pallet::weight(T::WeightInfo::store_key(key.len() as u32))]
         pub fn store_key(
-            _origin: OriginFor<T>,
+            origin: OriginFor<T>,
             identifier: VerificationKeyIdentifier,
             key: Vec<u8>,
         ) -> DispatchResult {
-            Self::bare_store_key(identifier, key).map_err(|e| e.into())
+            Self::bare_store_key(origin, identifier, key).map_err(|e| e.into())
         }
 
         /// Deletes a key stored under `identifier` in `VerificationKeys` map.
         ///
-        /// Can only be called by a root account.
+        /// Returns the deposit locked. Can only be called by the key owner.
         #[pallet::call_index(1)]
-        #[pallet::weight(T::DbWeight::get().writes(1))]
+        #[pallet::weight(T::WeightInfo::delete_key(T::MaximumVerificationKeyLength::get()))]
         pub fn delete_key(
             origin: OriginFor<T>,
             identifier: VerificationKeyIdentifier,
         ) -> DispatchResult {
-            ensure_root(origin)?;
+            let who = ensure_signed(origin).map_err(|_| Error::<T>::BadOrigin)?;
+            let owner = VerificationKeyOwners::<T>::get(identifier)
+                .ok_or(Error::<T>::UnknownVerificationKeyIdentifier)?;
+
+            ensure!(who == owner, Error::<T>::NotOwner);
+
+            let deposit = VerificationKeyDeposits::<T>::take((&owner, &identifier)).unwrap(); // cannot fail since the key has owner and owner must have made a deposit
+            T::Currency::unreserve(&owner, deposit);
+
             VerificationKeys::<T>::remove(identifier);
-            Self::deposit_event(Event::VerificationKeyDeleted);
+            Self::deposit_event(Event::VerificationKeyDeleted(who, identifier));
             Ok(())
         }
 
         /// Overwrites a key stored under `identifier` in `VerificationKeys` map with a new value `key`
         ///
         /// Fails if `key.len()` is greater than `MaximumVerificationKeyLength`.
-        /// Can only be called by a root account.
+        /// Can only be called by the original owner of the key.
+        /// It will require the caller to lock up additional funds (if the new key occupies more storage)
+        /// or reimburse the difference if it is shorter in it's byte-length.
```

Comment:
```suggestion
        /// or reimburse the difference if it is shorter in its byte-length.
```

Reply:
Still pretty sure I'm correct: it will reimburse the difference if the new key is shorter - if _its_ byte-length is shorter.